DSharpPlus/Entities/Interaction/DiscordInteractionDataOption.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities
+{
+    /// <summary>
+    /// Represents parameters for interaction commands. 
+    /// </summary>
+    public class DiscordInteractionDataOption
+    {
+        /// <summary>
+        /// Gets the name of this interaction parameter.
+        /// </summary>
+        [JsonProperty("name")]
+        public string Name { get; internal set; }
+
+        /// <summary>
+        /// Gets the option type of this interaction parameter. 
+        /// <para>This can be cast to an <see langword="int"></see>, <see langword="bool"></see>, or <see langword="string"></see> depending on the <see cref="DiscordInteractionDataOption.Name"/></para>
```

Comment:
You should also be able to cast to a ulong if the type is user, role or channel

Reply:
Discord also sends a `long` and not an `int`, and will throw an error while casting.

Regarding the `ulong` values, it turns out discord is SUPER inconsistent with it. Running the exact same command in quick succession, I sometimes successfully casted as a ulong, but sometimes I got a casting error where apparently discord sent a string and not a ulong value for a user. I seem to only get a string for roles. This could also be a problem with Newtonsoft.Json, but I'm not entirely sure. It might be better to advise them to run something like `ulong.Parse(OptionType.ToString())` instead of casting to avoid these kind of errors for user, role and channel ids.

Reply:
Yeah I may have to put a custom converter for that then.